docs/design/submarine-server/architecture.md

```diff
@@ -0,0 +1,126 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+  -->
+## Motivation
+Due to the single responsibility of considering different components, and nowadays the core module depends the hadoop libs which can be setted optional.
+
+## Proposal
+ ```
+                                                                +---------------------+
+ +-----------+                                                  | +--------+   +----+ |
+ |           |                                                  | |runtime1+-->+job1| |
+ | workbench +---+    +---------------------------------+       | +--------+   +----+ |
+ |           |   |    | +------+ +--------------------+ |   +-->+ +--------+   +----+ |
+ +-----------+   |    | |      | | +------+ +-------+ | |   |   | |runtime2+-->+job2| |
+                 |    | |      | | | YARN | |  K8s  | | |   |   | +--------+   +----+ |
+ +-----------+   |    | |      | | +------+ +-------+ | |   |   |     YARN Cluster    |
+ |           |   |    | |      | |     submitter      | |   |   +---------------------+
+ |    CLI    +------->+ |      | +--------------------+ +---+
+ |           |   |    | | REST | +--------+ +---------+ |   |   +---------------------+
+ +-----------+   |    | |      | | TRANSL | | monitor | |   |   | +--------+   +----+ |
+                 |    | |      | +--------+ +---------+ |   |   | |        +-->+job1| |
+ +-----------+   |    | |      | +--------------------+ |   |   | |        |   +----+ |
+ |           |   |    | |      | |     JobManager     | |   +-->+ |operator|   +----+ |
+ |    SDK    +---+    | +------+ +--------------------+ |       | |        +-->+job2| |
+ |           |        +---------------------------------+       | +--------+   +----+ |
+ +-----------+                                                  |     K8s Cluster     |
+    client                          server                      +---------------------+
+ ```
+We propose to split the core module to two modules, CLI and server as shown in FIG. The client call the REST API to submit and retrieve the job info. 
```

Comment:
"the core module" -> "the original core module in the old layout"

Reply:
Clever description, I will modify the doc.